src/ProgressBar/SinglaRobin/VanillaJS/index.js

```diff
@@ -0,0 +1,42 @@
+(function () {
+    const barElement = document.getElementById('progress-bar');
+    const runButton = document.getElementById('run-button');
+    const spanElement = document.querySelector('#run-button span');
+    // Time is in seconds
+    const animationTime = 3;
+    const maxProgress = 100;
+    let progress = 0;
+    let run = 0;
+    let timerId = null;
+
+    const updateBarValue = () => {
+        if (progress >= 100) {
+            progress = 0;
+            barElement.value = 0;
+            clearInterval(timerId);
+            timerId = null;
+            run--;
+            if (run === 0) {
+                spanElement.innerHTML = '';
+            } else {
+                spanElement.innerHTML = run;
+                handleProgressBar();
+            }
+        } else {
+            progress += maxProgress / animationTime;
```

Comment:
this is leading to jittering animation of progress bar, but as per requirements, we need to show smooth animation of the progress